src/peft/tuners/vera/config.py

```diff
@@ -0,0 +1,155 @@
+# limitations under the License.
+
+import warnings
+from dataclasses import dataclass, field
+from typing import List, Optional, Union
+
+from peft.config import PeftConfig
+from peft.utils import PeftType
+
+
+@dataclass
+class VeraConfig(PeftConfig):
+    """
+    This is the configuration class to store the configuration of a [`VeraModel`].
+
+    Paper: https://arxiv.org/abs/2310.11454.
+
+    Args:
+        r (`int`):
+            Vera attention dimension.
+        target_modules (`Union[List[str],str]`):
+            The names of the modules to apply Vera to. Only linear layers are supported.
+        projection_prng_key (`int`):
+            Vera PRNG init key. Used for initialising vera_A and vera_B for new models or when loading a checkpoint
+            that did not include these projections. Defaults to `0`.
+        save_projection (`bool`):
+            Whether to save the vera_A / vera_B projections in the state dict alongside per layer lambda_b / lambda_d
+            weights. This will increase the size of the checkpoint, but guarantee that we can reload the checkpoint on
+            all system configurations. Defaults to `True`.
+        vera_dropout (`float`):
+            The dropout probability for Vera layers.
+        d_initial (`float`):
+            Initial init value for `vera_lambda_d` vector used when `init_weights`.
+        fan_in_fan_out (`bool`):
+            Set this to True if the layer to replace stores weight like (fan_in, fan_out). For example, gpt-2 uses
+            `Conv1D` which stores weights like (fan_in, fan_out) and hence this should be set to `True`.
+        bias (`str`):
+            Bias type for Vera. Can be 'none', 'all' or 'vera_only'. If 'all' or 'vera_only', the corresponding biases
+            will be updated during training. Be aware that this means that, even when disabling the adapters, the model
+            will not produce the same output as the base model would have without adaptation.
+        modules_to_save (`List[str]`):
+            List of modules apart from Vera layers to be set as trainable and saved in the final checkpoint.
+        init_weights (`bool`):
+            Whether to initialize the weights of the Vera layers with their default initialization. Don't change this
+            setting, except if you know exactly what you're doing.
+        layers_to_transform (`Union[List[int],int]`):
+            The layer indexes to transform, if this argument is specified, it will apply the Vera transformations on
+            the layer indexes that are specified in this list. If a single integer is passed, it will apply the Vera
+            transformations on the layer at this index.
+        layers_pattern (`str`):
+            The layer pattern name, used only if `layers_to_transform` is different from `None` and if the layer
+            pattern is not in the common layers pattern.
+    """
+
+    r: int = field(default=8, metadata={"help": "Vera attention dimension"})
```

Comment:
perhaps we should increase the default value? 8 is rather small for VeRA (paper used 256-1024 for their experiments)